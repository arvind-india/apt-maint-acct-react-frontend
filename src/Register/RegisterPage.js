import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import {
          Form,
          Button,
          FormGroup,
          FormText,
          Input,
          Label,
          Col
        } from 'reactstrap'

import { userActions, alertActions } from '../_actions'
import { SocialLoginPage } from '../Login'

class RegisterPage extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      user: {
        name: '',
        first_name: '',
        last_name: '',
        email: '',
        password: '',
        infos: []
      },
      confirmPassword: '',
      submitted: false
    }
    this.handleChange = this.handleChange.bind(this)
    this.handleConfirmPasswordChange = this.handleConfirmPasswordChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }
  handleChange(event) {
    const { name, value } = event.target
    const { user } = this.state
    this.setState({
      user: {
        ...user,
        [name]: value
      }
    })
  }
  handleConfirmPasswordChange(event) {
    this.setState({
      confirmPassword: event.target.value
    })
  }
  handleSubmit(event) {
    event.preventDefault()
    this.setState({ submitted: true })
    const { user, confirmPassword } = this.state
    const { dispatch } = this.props
    if(user.name &&
        user.first_name &&
        user.last_name &&
        user.email &&
        user.password &&
        confirmPassword &&
        user.password === confirmPassword
      ) {
          dispatch(userActions.register(user))
        } else {
          dispatch(alertActions.error('Missing data...'))
        }
  }
  render() {
    // const { user, submitted, confirmPassword } = this.state
    const { alert } = this.props
    return (
      <div>
        <h2 align="center">Register</h2>
        {alert.message && <div className={`alert ${alert.type}`}>{alert.message}</div>}
        <Form onSubmit={this.handleSubmit}>
          { this.userName() }
          { this.firstName() }
          { this.lastName() }
          { this.password() }
          { this.repeatPassword() }
          { this.buttons() }
        </Form>
      </div>
    )
  }

  userName() {
    const { submitted, user } = this.state
    return <FormGroup row>
      <Label sm={{size: 1, offset: 4}}>UserName</Label>
      <Col sm={3}>
        <Input
          type="text"
          name="name"
          placeholder="User name here"
          onChange={this.handleChange}
        />
        {submitted && !user.name && <FormText color="danger">UserName is required</FormText>}
      </Col>
    </FormGroup>
  }
  firstName() {
    const { submitted, user } = this.state
    return <FormGroup row>
      <Label sm={{size: 1, offset: 4}}>FirstName</Label>
      <Col sm={3}>
        <Input
          type="text"
          name="first_name"
          placeholder="First name here"
          onChange={this.handleChange}
        />
        {submitted && !user.first_name && <FormText color="danger">First Name is required</FormText>}
      </Col>
    </FormGroup>
  }
  lastName() {
    const { submitted, user } = this.state
    return <FormGroup row>
      <Label sm={{size: 1, offset: 4}}>LastName</Label>
      <Col sm={3}>
        <Input
          type="text"
          name="last_name"
          placeholder="Last name here"
          onChange={this.handleChange}
        />
        {submitted && !user.last_name && <FormText color="danger">Last Name is required</FormText>}
      </Col>
    </FormGroup>
  }

  email() {
    const { submitted, user } = this.state
    return <FormGroup row>
      <Label sm={{size: 1, offset: 4}}>Email_Id</Label>
      <Col sm={3}>
        <Input
          type="email"
          name="email"
          placeholder="email id here"
          onChange={this.handleChange}
        />
        {submitted && !user.email && <FormText color="danger">Email-id is required</FormText>}
      </Col>
    </FormGroup>
  }
  password() {
    const { submitted, user } = this.state
    return <FormGroup row>
      <Label sm={{size: 1, offset: 4}}>Password</Label>
      <Col sm={3}>
        <Input
          type="password"
          name="password"
          placeholder="password here"
          onChange={this.handleChange}
        />
        {submitted && !user.password && <FormText color="danger">Password is required</FormText>}
      </Col>
    </FormGroup>
  }
  repeatPassword() {
    const { submitted, user, confirmPassword } = this.state
    return <FormGroup row>
      <Label sm={{size: 1, offset: 4}} title="Repeat password">Repeat_Pwd</Label>
      <Col sm={3}>
        <Input
          type="password"
          name="confirmPassword"
          placeholder="Repeat password here"
          onChange={this.handleConfirmPasswordChange}
        />
        {submitted && !confirmPassword && <FormText color="danger">Repeat Password is required</FormText>}
        {user.password && confirmPassword && user.password !== confirmPassword && <FormText color="danger">Password do NOT matchG</FormText>}
      </Col>
    </FormGroup>
  }

  buttons() {
    return <FormGroup row>
      <Col sm={{size: 3, offset: 5}}>
        <Button type="submit" color="primary" title="Register New User">Register</Button>
        <Button color="link">
          <Link to="/login" className="text-danger" title="Go to home">Cancel</Link>
        </Button>
        <SocialLoginPage />
      </Col>
    </FormGroup>
  }


}

function mapStateToProps(state) {
  const { alert } = state
  const { registering } = state.registration
  return {
    alert,
    registering
  }
}

const connectedRegisterPage = connect(mapStateToProps)(RegisterPage)
export { connectedRegisterPage as RegisterPage }
