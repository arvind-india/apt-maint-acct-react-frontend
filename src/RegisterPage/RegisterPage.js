import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import {
          Form,
          Button,
          FormGroup,
          FormFeedback,
          Input,
          Label,
          Col
        } from 'reactstrap'

import { userActions } from '../_actions'

class RegisterPage extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      user: {
        name: '',
        first_name: '',
        last_name: '',
        email: '',
        password: '',
        infos: []
      },
      confirmPassword: '',
      submitted: false
    }
    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }
  handleChange(event) {
    const { name, value } = event.target
    const { user } = this.state
    this.setState({
      user: {
        ...user,
        [name]: value
      }
    })
  }
  handleConfirmPasswordChange(event) {
    this.setState({
      confirmPassword: event.target
    })
  }
  handleSubmit(event) {
    event.preventDefault()
    this.setState({ submitted: true })

    const { user, confirmPassword } = this.state
    const { dispatch } = this.props
    if(user.name &&
        user.first_name &&
        user.last_name &&
        user.email &&
        user.password &&
        confirmPassword &&
        user.password === confirmPassword
      ) {
          dispatch(userActions.register(user))
        }
  }
  render() {

    const { user, submitted, confirmPassword } = this.state

    return (

      <div>

        <h2 align="center">Register</h2>

        <Form onSubmit={this.handleSubmit}>

          <FormGroup row>
            <Label sm={3}>UserName</Label>
            <Col sm={9}>
              <Input
                type="text"
                name="name"
                placeholder="User name here"
                defaultValue={user.name}
                onChange={this.handleChange}
              />
            </Col>
            <Col smoffset={3} sm={9}>
              {submitted && !user.name && <FormFeedback>User Name is required</FormFeedback>}
            </Col>
          </FormGroup>

          <FormGroup row>
            <Label sm={3}>FirstName</Label>
            <Col sm={9}>
              <Input
                type="text"
                name="first_name"
                placeholder="First name here"
                defaultValue={user.first_name}
                onChange={this.handleChange}
              />
            </Col>
            <Col smoffset={3} sm={9}>
              {submitted && !user.first_name && <FormFeedback>First Name is required</FormFeedback>}
            </Col>
          </FormGroup>

          <FormGroup row>
            <Label sm={3}>LastName</Label>
            <Col sm={9}>
              <Input
                type="text"
                name="last_name"
                placeholder="Last name here"
                defaultValue={user.last_name}
                onChange={this.handleChange}
              />
            </Col>
            <Col smoffset={3} sm={9}>
              {submitted && !user.last_name && <FormFeedback>Last Name is required</FormFeedback>}
            </Col>
          </FormGroup>

          <FormGroup row>
            <Label sm={3}>Email_Id</Label>
            <Col sm={9}>
              <Input
                type="email"
                name="email"
                placeholder="email id here"
                defaultValue={user.email}
                onChange={this.handleChange.bind(this)}
              />
            </Col>
            <Col smoffset={3} sm={9}>
              {submitted && !user.email && <FormFeedback>Email-id is required</FormFeedback>}
            </Col>
          </FormGroup>

          <FormGroup row>
            <Label sm={3}>Password</Label>
            <Col sm={9}>
              <Input
                type="password"
                name="password"
                placeholder="password here"
                defaultValue={user.password}
                onChange={this.handleChange.bind(this)}
              />
            </Col>
            <Col smoffset={3} sm={9}>
              {submitted && !user.password && <FormFeedback>Password is required</FormFeedback>}
            </Col>
          </FormGroup>

          <FormGroup row>
            <Label sm={3}>Repeat_Password</Label>
            <Col sm={9}>
              <Input
                type="password"
                name="confirmPassword"
                placeholder="Repeat password here"
                defaultValue={confirmPassword}
                onChange={this.handleConfirmPasswordChange.bind(this)}
              />
            </Col>
            <Col smoffset={3} sm={9}>
              {submitted && !confirmPassword && <FormFeedback>Repeat Password is required</FormFeedback>}
              {user.password && confirmPassword && user.password !== confirmPassword && <FormFeedback>Passwords are NOT MATCHING</FormFeedback>}
            </Col>
          </FormGroup>

          <FormGroup row>
            <Label sm={3}></Label>
            <Col smoffset={3} sm={9}>
              <Button type="submit" color="primary">Register</Button>
              <Button color="link">
                <Link to="/login">Cancel</Link>
              </Button>
            </Col>
          </FormGroup>

        </Form>

      </div>
    )
  }
}

function mapStateToProps(state) {
  const { registering } = state.registration
  return {
    registering
  }
}

const connectedRegisterPage = connect(mapStateToProps)(RegisterPage)
export { connectedRegisterPage as RegisterPage }
