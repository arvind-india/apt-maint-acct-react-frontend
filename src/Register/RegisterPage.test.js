// RegisterPage.test.js

import React from 'react'
import { shallow } from 'enzyme'

import { Register } from './RegisterPage'

const mockUsers = [
  {
    //id: 1,
    name: 'user1',
    first_name:'user1',
    last_name: 'test',
    email: 'user1@test.com',
    password: 'secret',
    infos: []
  },
  {
    // id: 2,
    name: 'user2',
    first_name:'user2',
    last_name: 'test',
    email: 'user2@test.com',
    password: 'secret',
    infos: []
  }

]

const mockProps = [
  {
    alert: {message: "testingAlert"},
    register: jest.fn(),
    error: jest.fn()
  },
  {
    alert: {message: "testingAlert"},
    register: jest.fn(),
    error: jest.fn()
  }
]
const mockLocations = [
  {
    state: {
      model: mockUsers[0],
      title: 'Mock Test: Registration1',
      module: '',
      password: 'secret',
      confirmPassword: 'secret',
      submitted: false
    }
  },
  {
    state: {
      model: mockUsers[1],
      title: 'Mock Test: Registration2',
      module: '',
      password: 'secret',
      confirmPassword: 'secret',
      submitted: false
    }
  }
]

const mocks = [
  {
    userName: {name: 'name', value: mockUsers[0].name},
    firstName: {name: 'first_name', value: mockUsers[0].first_name},
    lastName: {name: 'last_name', value: mockUsers[0].last_name},
    email: {name: 'email', value: mockUsers[0].email },
    password: {name: 'password', value: mockUsers[0].password}
  },
  {
    userName: {name: 'name', value: mockUsers[1].name},
    firstName: {name: 'first_name', value: mockUsers[1].first_name},
    lastName: {name: 'last_name', value: mockUsers[1].last_name},
    email: {name: 'email', value: mockUsers[1].email },
    password: {name: 'password', value: mockUsers[1].password}
  }
]
const component = shallow(<Register {...mockProps[0]} location={mockLocations[0]}/>)

describe('UsersPage params test', () => {
  it('should render UsersPage without crashing', () => {
    component
  })
  it('should render initial layout', () => {
    expect(component.getElements()).toMatchSnapshot()
  })
  it('should have model state', () => {
    expect(component.state('user')).toBeDefined()
  })
  it('should have confirmPassword state', () => {
    expect(component.state('confirmPassword')).toBeDefined()
  })
  it('should have submitted state', () => {
    expect(component.state('submitted')).toBeDefined()
  })

})


describe('RegisterPage events test1', () => {
  it('should call register on submit and no crash', () => {
    component.find('#registerForm').simulate('submit', {preventDefault() {}})
    expect(mockProps[0].register.mock.calls.length).toEqual(1)
  })
  it('should be called with required states as arguments', () => {
    component.find('#userName').simulate('change', {target: mocks[0].userName})
    component.find('#firstName').simulate('change', {target: mocks[0].firstName})
    component.find('#lastName').simulate('change', {target: mocks[0].lastName})
    component.find('#email').simulate('change', {target: mocks[0].email})
    component.find('#password').simulate('change', {target: mocks[0].password})
    component.find('#registerForm').simulate('submit', {preventDefault() {}})
    expect(mockProps[0].register.mock.calls[1][0]).toEqual(mockUsers[0])
  })
})


const component2 = shallow(<Register {...mockProps[1]} location={mockLocations[1]}/>)
describe('RegisterPage events test2', () => {
  it('should be called with required states as arguments', () => {
    component2.find('#userName').simulate('change', {target: mocks[1].userName})
    component2.find('#firstName').simulate('change', {target: mocks[1].firstName})
    component2.find('#lastName').simulate('change', {target: mocks[1].lastName})
    component2.find('#email').simulate('change', {target: mocks[1].email})
    component2.find('#password').simulate('change', {target: mocks[1].password})
    component2.find('#registerForm').simulate('submit', {preventDefault() {}})
    expect(mockProps[1].register.mock.calls[0][0]).toEqual(mockUsers[1])
  })
})
