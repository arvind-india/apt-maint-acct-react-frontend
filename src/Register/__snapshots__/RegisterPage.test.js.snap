// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersPage params test should render initial layout 1`] = `
Array [
  <div>
    <h2
        align="center"
    >
        Register
    </h2>
    <FlashMessage
        color={undefined}
        delay={4000}
        text="testingAlert"
    />
    <Form
        id="registerForm"
        onSubmit={[Function]}
        tag="form"
    >
        <FormGroup
            row={true}
            tag="div"
        >
            <Col
                sm={
                    Object {
                        "offset": 4,
                        "size": 4,
                      }
                }
                tag="div"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            >
                <Label
                    for="userName"
                    size="sm"
                    tag="label"
                    widths={
                        Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                    }
                >
                    User name
                </Label>
                <Input
                    id="userName"
                    name="name"
                    onChange={[Function]}
                    size="sm"
                    tag="p"
                    title="User name here"
                    type="text"
                />
            </Col>
        </FormGroup>
        <FormGroup
            row={true}
            tag="div"
        >
            <Col
                sm={
                    Object {
                        "offset": 4,
                        "size": 4,
                      }
                }
                tag="div"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            >
                <Label
                    for="email"
                    size="sm"
                    tag="label"
                    widths={
                        Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                    }
                >
                    Email id
                </Label>
                <Input
                    id="email"
                    name="email"
                    onChange={[Function]}
                    size="sm"
                    tag="p"
                    title="email id here"
                    type="email"
                />
            </Col>
        </FormGroup>
        <FormGroup
            row={true}
            tag="div"
        >
            <Col
                sm={
                    Object {
                        "offset": 4,
                        "size": 2,
                      }
                }
                tag="div"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            >
                <Label
                    for="firstName"
                    size="sm"
                    tag="label"
                    widths={
                        Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                    }
                >
                    First name
                </Label>
                <Input
                    id="firstName"
                    name="first_name"
                    onChange={[Function]}
                    size="sm"
                    tag="p"
                    title="First name here"
                    type="text"
                />
            </Col>
            <Col
                sm={
                    Object {
                        "size": 2,
                      }
                }
                tag="div"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            >
                <Label
                    for="lastName"
                    size="sm"
                    tag="label"
                    widths={
                        Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                    }
                >
                    Last name
                </Label>
                <Input
                    id="lastName"
                    name="last_name"
                    onChange={[Function]}
                    size="sm"
                    tag="p"
                    title="Last name here"
                    type="text"
                />
            </Col>
        </FormGroup>
        <FormGroup
            row={true}
            tag="div"
        >
            <Col
                sm={
                    Object {
                        "offset": 4,
                        "size": 2,
                      }
                }
                tag="div"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            >
                <Label
                    for="password"
                    size="sm"
                    tag="label"
                    widths={
                        Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                    }
                >
                    Password
                </Label>
                <Input
                    id="password"
                    name="password"
                    onChange={[Function]}
                    size="sm"
                    tag="p"
                    title="password here"
                    type="password"
                />
            </Col>
            <Col
                sm={
                    Object {
                        "size": 2,
                      }
                }
                tag="div"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            >
                <Label
                    for="confirmPassword"
                    size="sm"
                    tag="label"
                    widths={
                        Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                    }
                >
                    Repeat password
                </Label>
                <Input
                    id="confirmPassword"
                    name="confirmPassword"
                    onChange={[Function]}
                    size="sm"
                    tag="p"
                    title="Repeat password here"
                    type="password"
                />
                
            </Col>
        </FormGroup>
        <FormGroup
            row={true}
            tag="div"
        >
            <Col
                sm={
                    Object {
                        "offset": 4,
                        "size": 4,
                      }
                }
                tag="div"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            >
                <Button
                    color="primary"
                    disabled={true}
                    tag="button"
                    title="Missing \\"Required Data\\"..."
                    type="submit"
                >
                    Register
                </Button>
                <Button
                    color="link"
                    tag="button"
                >
                    <Link
                        className="text-danger"
                        replace={false}
                        title="Go to home"
                        to="/login"
                    >
                        Cancel
                    </Link>
                </Button>
                <Connect(SocialLogin) />
            </Col>
        </FormGroup>
    </Form>
</div>,
]
`;
